src/torch_fourier_filter/whitening.py

```diff
@@ -3,65 +3,90 @@
 import einops
 import torch
 import torch.nn.functional as F
-
-from torch_fourier_filter.dft_utils import (
-    _1d_to_rotational_average_2d_dft,
-    _1d_to_rotational_average_3d_dft,
-    bin_or_interpolate_to_output_size,
-    rotational_average_dft_2d,
-    rotational_average_dft_3d,
-)
+from torch_grid_utils.fftfreq_grid import fftfreq_grid
 
 
 def gaussian_smoothing(
-    tensor: torch.Tensor, kernel_size: int = 5, sigma: float = 1.0
+    tensor: torch.Tensor,
+    spatial_dims: int = 1,
```

Comment:
I think modifying how this function parses in arguments agree better with PyTorch's API makes some sense. Rather than the `spatial_dims` argument defining how many spatial dimensions the tensor has, use the `dims` keyword argument to define which dimensions smoothing should be applied along.

This can still be limited to tensors with one, two, or possibly three dimensions by asserting this `list`/`tuple` is at most length three.

Also does not require batch dimension to be along zeroth dimension